transaction/StronglyConnectedComponents.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_TRANSACTION_STRONGLY_CONNECTED_COMPONENTS_HPP_
+#define QUICKSTEP_TRANSACTION_STRONGLY_CONNECTED_COMPONENTS_HPP_
+
+#include <stack>
+#include <vector>
+#include <unordered_map>
+
+#include "transaction/DirectedGraph.hpp"
+#include "utility/Macros.hpp"
+
+namespace quickstep {
+
+namespace transaction {
+
+/** \addtogroup Transaction
+ *  @{
+ */
+
+/**
+ * @brief Class for applying Tarjan's strongly connected components
+ *        algorithm to the 
+ **/
+class StronglyConnectedComponents {
+ public:
+  /**
+   * @brief Constructor for StronglyConnectedComponents.
+   *
+   * @param directed_graph Const reference to the directed graph which the 
+   *        algorithm runs.
+   **/
+  explicit StronglyConnectedComponents(const DirectedGraph &directed_graph);
```

Comment:
Do you assume that there won't be any changes (e.g. addition or removal of a node) in ``directed_graph``? If not, then we need to provide methods which update the strongly connected components based on the changes in the original graph. In that case we will have to change the type of ``directed_graph_`` from const reference to a pointer.